packages/material-ui-lab/src/ClockPicker/ClockNumber.tsx

```diff
@@ -1,80 +1,114 @@
+  // TODO: spread to a `span`. What are the implications (generic role etc.)
+  'aria-label': string;
+  classes?: Partial<typeof clockNumberClasses>;
   disabled: boolean;
   index: number;
   inner: boolean;
   label: string;
   selected: boolean;
-  // TODO: spread to a `span`. What are the implications (generic role etc.)
-  'aria-label': string;
 }
 
-export type ClockNumberClassKey = 'root' | 'selected' | 'disabled' | 'inner';
+export type ClockNumberClassKey = keyof typeof clockNumberClasses;
 
-export const styles: MuiStyles<ClockNumberClassKey> = (theme): StyleRules<ClockNumberClassKey> => ({
-  root: {
-    width: CLOCK_HOUR_WIDTH,
-    height: CLOCK_HOUR_WIDTH,
-    position: 'absolute',
-    left: `calc((100% - ${CLOCK_HOUR_WIDTH}px) / 2)`,
-    display: 'inline-flex',
-    justifyContent: 'center',
-    alignItems: 'center',
-    borderRadius: '50%',
-    color: theme.palette.text.primary,
-    '&:focused': {
-      backgroundColor: theme.palette.background.paper,
-    },
-    '&$selected': {
-      color: theme.palette.primary.contrastText,
-    },
-    '&$disabled': {
-      pointerEvents: 'none',
-      color: theme.palette.text.disabled,
+export function getClockNumberUtilityClass(slot: string) {
+  return generateUtilityClass('MuiInternalClockNumber', slot);
+}
+
+export const clockNumberClasses = generateUtilityClasses('MuiInternalClockNumber', [
+  'root',
+  'selected',
+  'disabled',
+  'inner',
+]);
+
+const useUtilityClasses = (styleProps: ClockNumberProps) => {
+  const { inner, selected, disabled, classes } = styleProps;
+
+  const slots = {
+    root: ['root', inner && 'inner', selected && 'selected', disabled && 'disabled'],
+  };
+
+  return composeClasses(slots, getClockNumberUtilityClass, classes);
+};
+
+const ClockNumberRoot = experimentalStyled(
+  'span',
+  {},
+  {
+    name: 'MuiInternalClockNumber',
+    slot: 'Root',
+    overridesResolver: (props, styles: Record<ClockNumberClassKey, object>) => {
+      const { styleProps } = props;
+      return {
+        ...styles.root,
+        ...(styleProps.inner && styles.inner),
+      };
     },
   },
-  selected: {},
-  disabled: {},
-  inner: {
-    ...theme.typography.body2,
-    color: theme.palette.text.secondary,
+)(({ theme, styleProps = {} }) => ({
+  width: CLOCK_HOUR_WIDTH,
+  height: CLOCK_HOUR_WIDTH,
+  position: 'absolute',
+  left: `calc((100% - ${CLOCK_HOUR_WIDTH}px) / 2)`,
+  display: 'inline-flex',
+  justifyContent: 'center',
+  alignItems: 'center',
+  borderRadius: '50%',
+  color: theme.palette.text.primary,
+  '&:focused': {
+    backgroundColor: theme.palette.background.paper,
   },
-});
+  [`&.${clockNumberClasses.selected}`]: {
+    color: theme.palette.primary.contrastText,
+  },
+  [`&.${clockNumberClasses.disabled}`]: {
+    pointerEvents: 'none',
+    color: theme.palette.text.disabled,
+  },
+  ...(!!styleProps.inner && {
+    ...(theme.typography.body2 as React.CSSProperties),
```

Comment:
The cast is not required anymore now that https://github.com/mui-org/material-ui/pull/26228 was merged. Please update the remove.